src/test/scala/org/broadinstitute/dsde/workbench/sam/util/ShadowCasterSpec.scala

```diff
@@ -0,0 +1,45 @@
+package org.broadinstitute.dsde.workbench.sam.util
+import cats.effect.IO
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.concurrent.{ExecutionContext, Promise}
+import scala.concurrent.duration._
+
+class ShadowCasterSpec extends FlatSpec with Matchers with ScalaFutures {
+
+  "ShadowCaster" should "runWithShadow" in {
+
+    class TestCaster extends ShadowCaster {
+      override implicit val executionContext: ExecutionContext = scala.concurrent.ExecutionContext.global
+
+      val real = IO.pure(10)
+      val shadow = IO.timer(executionContext).sleep(1 second).map(_ => 15)
+
+      val resultPromise: Promise[(String, Any, Long, Any, Long)] = Promise()
+
+      def runTest(): IO[Int] = {
+        runWithShadow("runTest", real, shadow)
+      }
+
+      override protected def reportResult[T](functionName: String, realResult: T, realTime: Long, shadowResult: T, shadowTime: Long): Unit = {
```

Comment:
I think the report is critical functionality for the ShadowCaster.  I suggest not overriding this and testing it, possibly with more complex objects than `Int`.

Reply:
I need to override it in this test to test the other functionality... when we write a real report function we can test that specifically